pushy/src/main/java/com/turo/pushy/apns/ApnsChannelFactory.java

```diff
@@ -116,58 +101,47 @@ public void exceptionCaught(final ChannelHandlerContext context, final Throwable
+                        apnsClientHandler.gracefulShutdownTimeoutMillis(gracefulShutdownTimeoutMillis);
+                    }
+                }
+
                 final ChannelPipeline pipeline = channel.pipeline();
 
                 if (proxyHandlerFactory != null) {
                     pipeline.addFirst(proxyHandlerFactory.createProxyHandler());
                 }
 
-                final SslHandler sslHandler = sslContext.newHandler(channel.alloc());
-
-                sslHandler.handshakeFuture().addListener(new GenericFutureListener<Future<Channel>>() {
-                    @Override
-                    public void operationComplete(final Future<Channel> handshakeFuture) {
-                        if (handshakeFuture.isSuccess()) {
-                            final String authority = channel.remoteAddress().getHostName();
-
-                            final ApnsClientHandler.ApnsClientHandlerBuilder clientHandlerBuilder;
-
-                            if (signingKey != null) {
-                                clientHandlerBuilder = new TokenAuthenticationApnsClientHandler.TokenAuthenticationApnsClientHandlerBuilder()
-                                        .signingKey(signingKey)
-                                        .tokenExpirationMillis(tokenExpirationMillis)
-                                        .authority(authority)
-                                        .idlePingIntervalMillis(idlePingIntervalMillis);
-                            } else {
-                                clientHandlerBuilder = new ApnsClientHandler.ApnsClientHandlerBuilder()
-                                        .authority(authority)
-                                        .idlePingIntervalMillis(idlePingIntervalMillis);
-                            }
-
-                            if (frameLogger != null) {
-                                clientHandlerBuilder.frameLogger(frameLogger);
-                            }
-
-                            final ApnsClientHandler apnsClientHandler = clientHandlerBuilder.build();
-
-                            if (gracefulShutdownTimeoutMillis > 0) {
-                                apnsClientHandler.gracefulShutdownTimeoutMillis(gracefulShutdownTimeoutMillis);
-                            }
-
-                            pipeline.addLast(new FlushConsolidationHandler(FlushConsolidationHandler.DEFAULT_EXPLICIT_FLUSH_AFTER_FLUSHES, true));
-                            pipeline.addLast(new IdleStateHandler(idlePingIntervalMillis, 0, 0, TimeUnit.MILLISECONDS));
-                            pipeline.addLast(apnsClientHandler);
-                            pipeline.remove(ConnectionNegotiationErrorHandler.INSTANCE);
-
-                            channel.attr(CHANNEL_READY_PROMISE_ATTRIBUTE_KEY).get().trySuccess(channel);
-                        } else {
-                            tryFailureAndLogRejectedCause(channel.attr(CHANNEL_READY_PROMISE_ATTRIBUTE_KEY).get(), handshakeFuture.cause());
```

Comment:
This failure now makes its way to the `exceptionCaught` method in `ApnsClientHandler`.